File: queue.js
class Queue {
	constructor(){
		this.first = null;
		this.last = null;
	}
	
	dequeue(){
		if (this.first === null){
			return null
		}
		var n = this.first 

		if (n.next === null){
			this.last = null
			this.first = null
		}
		else {
			this.first = n.next
		}
		return n.data
	}
	
	enqueue(data){
		var n = new Node(data)
		if (this.last){
			this.last.next = n
			this.last = n
		}
		else {
			this.first = n
			this.last = n
		}
	}
}

class Node {
	constructor(data){
		this.data = data
		this.next = null
	}
}

q = new Queue()

File: queue.py
class Node:
	def __init__(self, val):
		self.val = val
		self.next = None
		
class Queue:
	first = None
	last = None
		
	def enqueue(self, val):
		n = Node(val)
		if self.last:
			self.last.next = n
			self.last = n
		else:
			self.first = n
			self.last = n 
			
	def dequeue(self):
		if self.first == None:
			return None
			
		n = self.first
		if n.next == None:
			self.last = None
			self.first = None
		else:
			self.first = n.next
		return n.val
		
		
queue = Queue()		
		